src/absenceDays/index.ts

```diff
@@ -1,24 +1,30 @@
 import BaseApi from '../baseApi';
 import responseHandler from '../utils/responseHandlers';
+import RequestParams from '../utils/requestParams';
 import { AbsenceDay } from './types';
 import { ReadParams } from '../utils/types';
 import { AxiosRequestConfig } from 'axios';
 
 export default class AbsenceDays extends BaseApi {
-  static resourceName = 'absencesDays';
+  private readonly resourceName = 'absencesDays';
 
-  public read(options: ReadParams = {}): Promise<AbsenceDay[]> {
-    const axiosConfig: AxiosRequestConfig = {
-      params: options,
-    };
-    const response = this._get<AbsenceDay[]>(`${AbsenceDays.resourceName}/read`, axiosConfig);
+  public read(requestParams?: RequestParams<AbsenceDay> | Object): Promise<AbsenceDay[]> {
+    const params = requestParams instanceof RequestParams ? requestParams?.getParams() : requestParams;
+    const response = this._get<AbsenceDay[]>(`${this.getResourceName()}/read`, { params });
     return responseHandler.requiredList(response);
   }
-  public readById(id: number, options: ReadParams = {}): Promise<AbsenceDay[]> {
-    const axiosConfig: AxiosRequestConfig = {
-      params: options,
-    };
-    const response = this._get<AbsenceDay[]>(`${AbsenceDays.resourceName}/read/${id}`, axiosConfig);
+
+  public readById(id: number, requestParams?: RequestParams<AbsenceDay>): Promise<AbsenceDay[]> {
```

Comment:
the same about RequestParams. if we agree to do this, then change needs to be done for all resource classes

Reply:
Topic for tea party